### DISCUSSION TITLE
setuptools==60.0.0 not removed trailing L

### DISCUSSION THREAD

**Discussion by Rokfordchez** [pypa/setuptools#2943] [Q&A] [answered]
It's okay that setuptools==60.0.0 not removed trailing L?

for example

1. upgrade setuptools from 59.8.0 to 60.0.0
2. install pycrypto that used trailing L in source
`pip3 --no-cache-dir install pycrypto~=2.6.1`

```
Python 3.7.4 (v3.7.4:e09359112e, Jul  8 2019, 14:54:52) 
[Clang 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from Crypto.Util.number import getRandomNBitInteger
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.7/dist-packages/Crypto/Util/number.py", line 139
    value |= 2L ** (N-1)                # Ensure high bit is set
              ^
SyntaxError: invalid syntax
```

with 59.8.0 everything fine and old libs like pycrypto==2.6.1 install correctly

**Comment by bikong2** [comment]
not work, any suggestions?

**Reply by bikong2** [reply]
i'm using: import timm

**Comment by abravalheri** [comment, marked as answer]
Hello,

Setuptools tools no longer supports Python 2 syntax.
No syntax conversion is performed during build.
